teamMaster.py

```diff
@@ -0,0 +1,264 @@
+"""
+teamMaster.py
+Author: Robert Becker
+Date: May 21, 2017
+Purpose: use daily schedule to extract team data from gameday boxscore
+
+Setup file/url names based on optional date arugment (mm-dd-yyyy), or today
+Load daily schedule dictionary into memory
+Use schedule to search thru boxscore dictionaries to get all team info
+Update teamMaster dictionary with any new team data found
+"""
+
+
+import sys
+import shutil
+import argparse
+import logging
+import logging.config
+import datetime
+import json
+import requests
+
+
+# setup global variables
+SCRIPT = 'teamMaster.py'
+LOGGING_INI = 'teamMaster_logging.ini'
+DAILY_SCHEDULE = 'Data/schedule_YYYYMMDD.json'
+TEAM_MSTR_I = 'Data/teamMaster.json'
+TEAM_MSTR_O = 'Data/teamMaster_YYYYMMDD.json'
+BOXSCORE = 'http://gd2.mlb.com/_directory_/boxscore.json'
+
+
+class LoadDictionaryError(ValueError):
+    pass
+
+
+def init_logger():
+    """
+    initialize global variable logger and set script name
+    """
+
+    global logger
+    logger = logging.getLogger(SCRIPT)
+
+
+def get_command_arguments():
+    """
+    --date  optional, will extract teams found in boxscore for this date
+    """
+
+    parser = argparse.ArgumentParser('teamMaster command line arguments')
+    parser.add_argument(
+        '-d',
+        '--date',
+        help='Date of daily schedule for input - mm-dd-yyyy format',
+        dest='game_date',
+        type=str)
+
+    argue = parser.parse_args()
+    logger.info('Command arguments: ' + str(argue.game_date))
+
+    return argue
+
+
+def determine_filenames(gamedate=None):
+    """
+    :param game_date: date of the games in format "MM-DD-YYYY"
+    """
+
+    if gamedate is not None:
+        yyyy = gamedate[6:10]
+        mm = gamedate[0:2]
+        dd = gamedate[3:5]
+    else:
+        yyyy = datetime.date.today().strftime("%Y")
+        mm = datetime.date.today().strftime("%m")
+        dd = datetime.date.today().strftime("%d")
+
+    mmddyyyy = mm + '-' + dd + '-' + yyyy
```

Comment:
Since I was getting today's date in pieces, that assignment statement was done to get the date in the right format (mm-dd-yyyy) for later purpose regardless if we are using today's date or what's passed as a command line argument. I can incorporate an assignment into the IF statement above it and would now look like this...

if gamedate is not None:
    mmddyyyy = gamedate
else:
    mmddyyyy = str(datetime.date.today())
    yyyymmdd = mmddyyyy[6:10] + mmddyyyy[0:2] + mmddyyyy[3:5]

schedule_input = DAILY_SCHEDULE.replace('YYYYMMDD', yyyymmdd)
team_output = TEAM_MSTR_O.replace('YYYYMMDD', yyyymmdd)

I don't see how I can use strftime to make the string when the date is passed as a string from the command line. Is this good?